From 6e231b2606e477c823874c54bb12f9edf47dbaa2 Mon Sep 17 00:00:00 2001
From: Arnaldo Garcia Rincon <agarrcia@amazon.com>
Date: Tue, 8 Apr 2025 17:42:23 +0000
Subject: [PATCH] Always restrict boringcrypto crypto/tls to FIPS

Rather than changing each Go package to enable FIPS mode, automatically
enable it only when the program is compiled with BoringCrypto.

Signed-off-by: Arnaldo Garcia Rincon <agarrcia@amazon.com>
---
 src/crypto/tls/boring.go  | 8 ++++++++
 src/go/build/deps_test.go | 5 ++---
 2 files changed, 10 insertions(+), 3 deletions(-)
 create mode 100644 src/crypto/tls/boring.go

diff --git a/src/crypto/tls/boring.go b/src/crypto/tls/boring.go
new file mode 100644
index 0000000..df778a1
--- /dev/null
+++ b/src/crypto/tls/boring.go
@@ -0,0 +1,8 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+//go:build boringcrypto
+
+package tls
+
+import _ "crypto/tls/fipsonly"
diff --git a/src/go/build/deps_test.go b/src/go/build/deps_test.go
index e3e0107..8e70a0a 100644
--- a/src/go/build/deps_test.go
+++ b/src/go/build/deps_test.go
@@ -550,8 +550,6 @@ var depsRules = `
 
 	FIPS, sync/atomic < crypto/tls/internal/fips140tls;
 
-	crypto/internal/boring/sig, crypto/tls/internal/fips140tls < crypto/tls/fipsonly;
-
 	CRYPTO, golang.org/x/sys/cpu, encoding/binary, reflect
 	< golang.org/x/crypto/internal/alias
 	< golang.org/x/crypto/internal/subtle
@@ -560,11 +558,12 @@ var depsRules = `
 	< golang.org/x/crypto/chacha20poly1305;
 
 	CRYPTO-MATH, NET, container/list, encoding/hex, encoding/pem,
-	golang.org/x/crypto/chacha20poly1305, crypto/tls/internal/fips140tls
+	golang.org/x/crypto/chacha20poly1305, crypto/internal/boring/sig, crypto/tls/internal/fips140tls
 	< crypto/internal/hpke
 	< crypto/x509/internal/macos
 	< crypto/x509/pkix
 	< crypto/x509
+	< crypto/tls/fipsonly
 	< crypto/tls;
 
 	# crypto-aware packages
-- 
2.49.0

